### ISSUE TITLE
Ollama custom URL with basic auth

### ISSUE THREAD

**Issue by agung2001** [logancyang/obsidian-copilot#467]
**Is your feature request related to a problem? Please describe.**
I would like to use a custom URL for my ollama instance from my server : `https://ollama.example.com`
but it uses a basic auth token for the authentication process

VSCode extension such as continue.dev uses the same configuration : https://github.com/continuedev/continue/issues/834#issuecomment-1953750202
```json
"models": [ {
      "title": "TinyLLAMA",
      "provider": "ollama",
      "model": "tinyllama",
      "apiBase": "https://ollama.example.com",
      "requestOptions": {
        "headers": {"Authorization": "Basic bXl1c2VybmFtZTpteXBhc3N3b3Jk"}
      }
    }]
}
```

**Describe the solution you'd like**
I think adding an additional option to configure the fetch function will be very helpful

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by logancyang** [comment]
This one seems related https://github.com/langchain-ai/langchainjs/issues/6631

**Comment by ivanvmoreno** [comment]
Seconding this request, https://github.com/langchain-ai/langchainjs/issues/6631 seems to be merged, unblocking this issue

**Comment by rubin110** [comment]
Any chance auth info can just be passed through the URL?

```
https://username:password123@ollama.example.com/
```

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
